spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/context/MessageSourceAutoConfiguration.java

```diff
@@ -81,6 +89,25 @@ public MessageSource messageSource(MessageSourceProperties properties) {
 		}
 		messageSource.setAlwaysUseMessageFormat(properties.isAlwaysUseMessageFormat());
 		messageSource.setUseCodeAsDefaultMessage(properties.isUseCodeAsDefaultMessage());
+
+		try {
+			if (StringUtils.hasText(properties.getCommonMessages())) {
+				PropertiesFactoryBean propertiesFactory = new PropertiesFactoryBean();
```

Comment:
We might be better off using `PropertiesLoaderUtils.fillProperties` here rather than `PropertiesFactoryBean`. Using a `FactoryBean` isn't ideal since we need to call `afterPropertiesSet` etc.